packages/privacy-toolset/src/cookie-banner/test/index.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * @jest-environment jsdom
+ */
+import { render, fireEvent, getByText, getByTestId } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { CookieBanner, CookieBannerProps } from '..';
+
+const genericContent: CookieBannerProps[ 'content' ] = {
+	simpleConsent: {
+		description: 'simple description',
+		acceptAllButton: 'acceptAllButton',
+		customizeButton: 'customizeButton',
+	},
+	customizedConsent: {
+		description: 'customized description',
+		categories: {
+			essential: {
+				name: 'essential name',
+				description: 'essential description',
+			},
+			analytics: {
+				name: 'analytics name',
+				description: 'analytics description',
+			},
+			advertising: {
+				name: 'advertising name',
+				description: 'advertising description',
+			},
+		},
+		acceptSelectionButton: 'acceptSelectionButton',
+	},
+};
+
+describe( 'CookieBanner', () => {
+	test( 'renders in simple consent correctly', () => {
+		const { container } = render(
+			<CookieBanner content={ genericContent } onAccept={ jest.fn } />
+		);
+		expect( container ).toMatchSnapshot();
```

Comment:
What do you think about checking for the presence of key controls here? Maybe seeing if the Accept and Customize buttons exist? The snapshot will tell us if any of the output changes, but we can always regenerate the snapshot to make the test pass.

Reply:
> but we can always regenerate the snapshot to make the test pass.
same as you can always change a test to make it pass. I think of them as good indicators you should pay attention to whether you're impacting something visually - to ensure the change is actually expected.

> Maybe seeing if the Accept and Customize buttons exist?
hmm, I'm not sure about it - it will not hurt to add these checks, but they are not necessary IMO

I was always thinking that in moments you need to check multiple visual elements (like the existence of paragraphs and buttons) - matching snapshot is a way to go. but I'm open to the discussion and improving these test cases so to make this component more bulletproof.

Reply:
> I was always thinking that in moments you need to check multiple visual elements (like the existence of paragraphs and buttons) - matching snapshot is a way to go

Yeah, I think snapshot tests are useful here for making sure future versions of the output don't change. Seems worth including here.

I've always thought of snapshot tests as fragile since any changes have to be manually checked when there are differences. It's possible that a developer could make a known change, expect the snapshot to be different and quickly regenerate the snapshot without realizing that their known change created another unknown effect. 

As you mentioned, it's also not super useful to check that a bunch of different sub-components are rendered/ exist. Especially when the main goal of a test is to prove that a component renders successfully. Perhaps we don't need to check for the Accept/ Customize buttons in this test since you are effectively testing that those buttons exist and are actionable in later tests. If this were the _only_ test in the file, I'd lean towards testing more than a snapshot, but I think just a snapshot is OK given that the later tests prove the function of the critical controls.

Reply:
I understand your point, thanks for sharing it. In that way, I'll actually add a test to ensure the critical controls are in order, and the rest of the elements I'll leave up to the snapshot test.